examples/jsm/renderers/common/RenderObject.js

```diff
@@ -69,19 +69,36 @@ export default class RenderObject {
 		const geometry = this.geometry;
 
 		const attributes = [];
+		const vertexBuffers = new Set();
 
 		for ( const nodeAttribute of nodeAttributes ) {
 
-			attributes.push( nodeAttribute.node && nodeAttribute.node.attribute ? nodeAttribute.node.attribute : geometry.getAttribute( nodeAttribute.name ) );
+			const attribute = nodeAttribute.node && nodeAttribute.node.attribute ? nodeAttribute.node.attribute : geometry.getAttribute( nodeAttribute.name );
+
+			attributes.push( attribute );
+
+			const bufferAttribute = attribute.isInterleavedBufferAttribute ? attribute.data : attribute;
+			vertexBuffers.add( bufferAttribute );
 
 		}
 
 		this.attributes = attributes;
+		this.vertexBuffers = Array.from( vertexBuffers.values() );
 
 		return attributes;
 
 	}
 
+	getVertexBuffers() {
+
+		if ( this.vertexBuffers !== null ) return this.vertexBuffers;
+
+		this.getAttributes();
```

Comment:
```suggestion
		if ( this.vertexBuffers === null ) this.getAttributes();
```

Reply:
Makes sense. I was echoing the pattern from getAttributes().
I also noticed I missed setting vertexBuffers to null which was in my working copy.